### ISSUE TITLE
Consider changing paragraph `lead` boolean attribute to appearance="lead"

### ISSUE THREAD

**Issue by Robbert** [nl-design-system/utrecht#1800]
`<utrecht-paragraph lead small>` would not work, these are conflicting. Perhaps those two should not be boolean attributes, but one enum.

So instead of:

- `<utrecht-paragraph lead>`
- `<utrecht-paragraph small>`

We'd have:

- `<utrecht-paragraph appearance="lead">`
- `<utrecht-paragraph appearance="small">`

In theory `appearance` could be implemented as a token list in the future (just like the "class" attribute), to allow for `<utrecht-paragraph appearance="small warning">`, then we would be back to square one.

**Comment by VincentSmedinga** [comment]
Yes, exclusive options work best with enums. This matches the Amsterdam approach, although we use a `size` prop with options `large` and `small`.

I think the broadness of the term ‘appearance’ makes it too easy to mix concerns and run into problems. The ‘warning’ connotation is a different aspect, deserving a separate prop.

Discriminating unions can be used in TypeScript to disallow unwanted combinations like a small warning. And in CSS, something like `color: hotpink` might come to the rescue :)

**Comment by remypar5** [comment]
I'm currently working on this. Everything is ready for making a PR, but there's some authentication issues so I cannot push my changes, yet.

**Comment by Robbert** [comment]
✅ React component

Nog even open tot alle frameworks gemigreerd zijn
